ebml/EbmlElement.h

```diff
@@ -293,6 +293,21 @@ class EBML_DLL_API EbmlSemanticContext {
 */
 class EBML_DLL_API EbmlElement {
   public:
+    // callback to tell if the element should be written or not
+    // \return true if the element should be written
+    using ShouldWrite = bool (*)(const EbmlElement &);
```

Comment:
I'd like to see this type changed to `std::function<bool(const EbmlElement &)>`. That's much more flexible than a old C-style function pointer, e.g. you can pass lambdas as an `std::function`, which would allow you to associate state with the function (what you considered using a separate `void*` argument for).

Reply:
I was planning on giving this a try but you beat me to it :) This is definitely better.

I cherry-picked your commits and rebased.